Cannot build tarantool statically with Clang 16 on Linux

**Bug description**

* OS: Linux
* OS Version: Ubuntu 20.04 
* Architecture: x86_64

Tarantool `3.1.0-entrypoint-71-gd29234ae07`.

**Steps to reproduce**

```
1. CC=clang-16 CXX=clang++-16 cmake . -DBUILD_STATIC_WITH_BUNDLED_LIBS=ON -DTEST_BUILD=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_WERROR=ON
2. make -j $(nproc)
```

**Actual behavior**

After execution of the second step, we will get the following error:

```console
...
[ 11%] Building C object src/CMakeFiles/cpu_feature.dir/cpu_feature.c.o
clang: error: argument unused during compilation: '-static-libstdc++' [-Werror,-Wunused-command-line-argument]
make[2]: *** [src/CMakeFiles/cpu_feature.dir/build.make:63: src/CMakeFiles/cpu_feature.dir/cpu_feature.c.o] Error 1
...
```

**Expected behavior**

The build is successful.

**Possible solution**

The aforementioned error can be easily fixed by applying the following patch:

```diff
diff --git a/cmake/CheckDependencies.cmake b/cmake/CheckDependencies.cmake
index 8745af401a..a9007396d6 100644
--- a/cmake/CheckDependencies.cmake
+++ b/cmake/CheckDependencies.cmake
@@ -28,6 +28,7 @@ elseif(UNIX)
         libgcc_s
         libpthread
         libsvace
+        libstdc++
     )
 else()
     message(FATAL_ERROR "Unknown platform")
diff --git a/cmake/compiler.cmake b/cmake/compiler.cmake
index 13b0229c02..c6cc031491 100644
--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -155,10 +155,10 @@ endif()
 option(ENABLE_BUNDLED_LIBUNWIND "Bundled libunwind will be built"
        ${ENABLE_BUNDLED_LIBUNWIND_DEFAULT})

-# On macOS there is no '-static-libstdc++' flag and it's use will
-# raise following error:
-# error: argument unused during compilation: '-static-libstdc++'
-if(BUILD_STATIC AND NOT TARGET_OS_DARWIN)
+# In Clang there is no '-static-libstdc++' flag and its use will raise
+# the following error:
+#     clang: error: argument unused during compilation: '-static-libstdc++'
+if(BUILD_STATIC AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # Static linking for c++ routines
     add_compile_flags("C;CXX" "-static-libstdc++")
 endif()
```

--- a/cmake/CheckDependencies.cmake
+++ b/cmake/CheckDependencies.cmake
@@ -28,6 +28,7 @@ elseif(UNIX)
         libgcc_s
         libpthread
         libsvace
+        libstdc++
     )
 else()
     message(FATAL_ERROR "Unknown platform")

--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -155,10 +155,10 @@ endif()
 option(ENABLE_BUNDLED_LIBUNWIND "Bundled libunwind will be built"
        ${ENABLE_BUNDLED_LIBUNWIND_DEFAULT})
 
-# On macOS there is no '-static-libstdc++' flag and it's use will
-# raise following error:
-# error: argument unused during compilation: '-static-libstdc++'
-if(BUILD_STATIC AND NOT TARGET_OS_DARWIN)
+# In Clang there is no '-static-libstdc++' flag and its use will raise
+# the following error:
+#     clang: error: argument unused during compilation: '-static-libstdc++'
+if(BUILD_STATIC AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     # Static linking for c++ routines
     add_compile_flags("C;CXX" "-static-libstdc++")
 endif()

